### PR TITLE
DSG-1397: App loading state

### PR DESCRIPTION
**Pull request by SouradeepSaha**
# Basics

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature
- [x] Styling and layout changes
- [ ] Code style changes (formatting, local variables)
- [ ] Refactoring (no functional changes)
- [ ] Build or CI related changes
- [ ] Tests
- [ ] Other... Please describe:

## What changes does this PR introduce?

Adds spinner functionality to different app components.

## Checklist

- [x] I have updated the documentation accordingly
- [x] I have confirmed all acceptance criteria defined by the story have been met
- [ ] I have confirmed that Sonar Qube findings for new code are reviewed and fixed, or will be whitelisted

### CODE REVIEW

File: libs/shared/feature/authentication/src/lib/components/authentication-email/authentication-email.component.html [hunk 1]
```diff
@@ -42,8 +42,10 @@ <h1 nuverialCardContentType="title">{{language.headerText}}</h1>
 
       <div class="dsg-authentication-email-content-controls">
         <nuverial-button
+          class="loading-button"
           [ariaLabel]="language.formSubmitText || ''"
           [disabled]="(submitStatus$ | async)?.status !== 'valid'"
+          [hasLoading]="true"
```
Comment (davidyoung77) [inline libs/shared/feature/authentication/src/lib/components/authentication-email/authentication-email.component.html:48]: We had removed has loading

File: libs/shared/ui/nuverial/src/lib/components/button/button.component.ts [hunk 2]
```diff
@@ -72,6 +72,10 @@ export class NuverialButtonComponent implements OnInit {
   /**
    * Button's event emitter for click events
    */
+  @Input() public hasLoading = false;
```
Comment (davidyoung77) [inline libs/shared/ui/nuverial/src/lib/components/button/button.component.ts:75]: We removed has loading, in favor of the class loading-button

File: libs/shared/ui/nuverial/src/lib/components/button/button.component.scss [hunk 3]
```diff
@@ -17,6 +17,13 @@
   border-radius: rem.fromPx(6px);
   display: inline-flex;
 
+  &.nuverial-button {
+    .hasLoading {
+      padding-left: rem.fromPx(50px);
+      padding-right: rem.fromPx(50px);
+    }
```
Comment (davidyoung77) [inline libs/shared/ui/nuverial/src/lib/components/button/button.component.scss:24]: I believe this can be removed

**Review by davidyoung77 — CHANGES_REQUESTED** [review]

**Comment by SouradeepSaha** [comment]
Closed PR for: https://github.com/Nuvalence/dsgov-web/pull/458

**Pull request closed without merging**